src/modules/security.js

```diff
@@ -512,11 +509,82 @@
       errorMessageKey: 'requiredFields'
     });
 
+    /*
+     * Google reCaptcha 2
+     */
+    $.formUtils.addValidator({
+        name: 'recaptcha',
+        validatorFunction: function (val, $el)
+        {
+            return grecaptcha.getResponse($el.data('validation-recaptcha-widget-id'));
+        },
+        errorMessage: '',
+        errorMessageKey: 'badreCaptcha'
+    });
 
-
-  $.fn.displayPasswordStrength = function(conf) {
+    $.fn.displayPasswordStrength = function(conf) {
         new $.formUtils.validators.validate_strength.strengthDisplay(this, conf);
         return this;
     };
 
+    var setupGooglereCaptcha = function (evt, $forms, config)
+    {
+        if( typeof grecaptcha !== typeof undefined ){
+            throw new Error('reCaptcha API can not be loaded by hand, delete reCaptcha API snippet.');
+        }
+
+        var src = '//www.google.com/recaptcha/api.js?onload=reCaptchaLoaded&render=explicit' + (config.lang ? '&hl=' + config.lang : '');
+        if ($('body').find('script[src="' + src + '"]').length === 0)
+        {
```

Comment:
Does this if-statement make any sense? You have already checked that grecaptcha is undefined.